Testable example breaks unit tests (false pass)

One of the supplied testable examples breaks unit tests:

```go
func Example() {
	check := nagiosplugin.NewCheck()
	// Make sure the check always (as much as possible) exits with
	// the correct output and return code if we terminate unexpectedly.
	defer check.Finish()
	// [...]
	// Output:
	// OK: Everything looks shiny from here, cap'n | badness=3.141592653589793;4000;9000;0;
}
```

`check.Finish()` invokes `os.Exit()`.  The test runner is killed before it is able to report on the final test result.

To see this in action, let's add an intentionally broken unit test and run `go test`.

```
diff --git a/range_test.go b/range_test.go
index 32c10ec..1e7dda0 100644
--- a/range_test.go
+++ b/range_test.go
@@ -162,3 +162,7 @@ func TestNegInfRange(t *testing.T) {
                }
        }
 }
+
+func TestBrokenThing(t *testing.T) {
+       t.Errorf("THE SKY IS FALLING")
+}
```

```
saj@sundown % go test
--- FAIL: TestBrokenThing (0.00s)
        range_test.go:167: THE SKY IS FALLING
ok      github.com/olorin/nagiosplugin     0.010s
saj@sundown % echo $?
0
```

Removing the testable examples results in expected behaviour:

```
saj@sundown % git status -s
 D example_test.go
 M range_test.go
saj@sundown % go test
--- FAIL: TestBrokenThing (0.00s)
        range_test.go:167: THE SKY IS FALLING
FAIL
exit status 1
FAIL    github.com/olorin/nagiosplugin     0.007s
saj@sundown % echo $?
1
```

I'm not sure how best to approach this problem in a testable example.

--- a/example_test.go
+++ b/example_test.go
@@ -30,6 +30,6 @@ func Example() {
 	// if there's more than one, the highest result will be the one
 	// returned (in ascending order OK, WARNING, CRITICAL, UNKNOWN).
 
-	// Output:
+	// This will print:
 	// OK: Everything looks shiny from here, cap'n | badness=3.141592653589793;4000;9000;0;
 }

